pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java

```diff
@@ -173,51 +179,67 @@ public static RestHighLevelClient getClientInstance() {
+                return new RestHighLevelClient(
+                        RestClient.builder(new HttpHost(url.getHost(), url.getPort(), "http")));
+            }
 
-        return new RestHighLevelClient(builder);
+            final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(userName, password));
+
+            URL proxyUrl = StringUtils.isNotBlank(proxy) ? new URL(proxy) : null;
+            if (proxyUrl != null) {
+                logger.info("Using Elasticsearch client proxy: {}", proxyUrl);
+            }
+
+            RestClientBuilder builder = RestClient.builder(new HttpHost(url.getHost(), url.getPort(), url.getProtocol()))
+                    .setHttpClientConfigCallback(httpClientBuilder -> {
+                        httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+                        if (proxyUrl != null) {
+                            httpClientBuilder.setProxy(new HttpHost(proxyUrl.getHost(), proxyUrl.getPort(), proxyUrl.getProtocol()));
+                            httpClientBuilder.setConnectionReuseStrategy(NoConnectionReuseStrategy.INSTANCE);
+                            httpClientBuilder.setDefaultIOReactorConfig(IOReactorConfig.custom().setSoKeepAlive(true).build());
+                        }
+                        return httpClientBuilder;
+                    });
+
+            return new RestHighLevelClient(builder);
+        } catch (MalformedURLException e) {
+            throw new DsmInternalError("Invalid ES client URL: " + baseUrl, e);
+        }
     }
 
-    public static RestHighLevelClient getClientForElasticsearchCloudCF(@NonNull String baseUrl, @NonNull String userName,
-                                                                       @NonNull String password, String proxy)
-            throws MalformedURLException {
-        final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
-        credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(userName, password));
+    private static String getProxy() {
+        return DSMConfig.hasConfigPath(ApplicationConfigConstants.ES_PROXY)
+                ? DSMConfig.getSqlFromConfig(ApplicationConfigConstants.ES_PROXY) : null;
+    }
 
-        URL url = new URL(baseUrl);
+    public static RestHighLevelClient getClientForElasticsearchCloudCF(@NonNull String baseUrl, @NonNull String userName,
+                                                                       @NonNull String password, String proxy) {
         return getClientForElasticsearchCloud(baseUrl, userName, password, proxy);
     }
 
+    protected static SearchResponse search(SearchRequest searchRequest) {
+        try {
+            SearchResponse res = client.search(searchRequest, RequestOptions.DEFAULT);
+            logger.info("ES search returned {}", res.getHits().getTotalHits());
```

Comment:
Maybe add "hits" to the log entry so readers know it's units and not a return code.

Reply:
Aha, in that case, never mind.